addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHttpMethodScanRuleUnitTest.java

```diff
@@ -34,6 +38,36 @@ protected InsecureHttpMethodScanRule createScanner() {
         return new InsecureHttpMethodScanRule();
     }
 
+    @ParameterizedTest
+    @CsvSource({"PUT, json", "PUT, xml", "PATCH, json", "PATCH xml"})
+    void shouldRaiseNoAlertsForPutOrPatchMethodsIfReturnJsonOrXml(String method, String contentType)
+            throws Exception {
+        // Given
+        HttpMessage message = new HttpMessage();
+        message.setRequestHeader(method + " /test HTTP/1.1 \r\nHost: www.example.com");
```

Comment:
The tests of active scan rules should not use external domains, only the local test server(s). (Unless there's a strong reason but it's not the case here.)